### ISSUE TITLE
Populate not working within Model.find call

### ISSUE THREAD

**Issue by ThoseGrapefruits** [Automattic/mongoose#11684]
**Do you want to request a *feature* or report a *bug*?**
Report a bug

**What is the current behavior?**
When calling something like
```mjs
  Model.find({}, null, { populate: 'fieldname' });
```

The fieldname is not populated.

**If the current behavior is a bug, please provide the steps to reproduce.**

Save as `repro.mjs` and run with `node repro.mjs`. In Mongoose `6.3.0`, it incorrectly prints an `ObjectId` for the `friend`. In Mongoose `6.2.7`, it correctly prints the fully populated `Cat` instance.
```mjs
import mongoose from 'mongoose';

mongoose.connect('mongodb://localhost:27017/cats', {
  useNewUrlParser: true, useUnifiedTopology: true
});

const Cat = mongoose.model('Cat', {
  name: String, 

  friend: {
    type: mongoose.ObjectId,
    ref: 'Cat'
  }
});

const friend = new Cat({ name: 'Floof' });
await friend.save();

const myCat = new Cat({ name: 'Lord Fluffles', friend: friend.id })
await myCat.save();

let cats = await Cat.find({}, null, { populate: 'friend' });

console.log('--populated-friends--', ...cats
  .map(cat => cat.friend)
  .filter(friend => friend));

await Cat.deleteMany({});

process.exit(0);
```

**What is the expected behavior?**
It should populate the `friend` field on all `Cat` documents returned by the `find` call.

**What are the versions of Node.js, Mongoose and MongoDB you are using? Note that "latest" is not a version.**

Node.js: `17.7.2`
Mongoose: `6.3.0`
MongoDB: `5.0.6`

**Comment by IslandRhythms** [comment]
https://mongoosejs.com/docs/api.html#query_Query-setOptions
`populate` is not an option for `find()`

**Comment by ThoseGrapefruits** [comment]
It is, but it's in a separate section. We've been using it (in production) for multiple years, and it works as of mongoose `6.2.7`. Screenshot of the docs section:

<img width="733" alt="Screen Shot 2022-04-19 at 11 54 35 " src="https://user-images.githubusercontent.com/1885875/164075580-a4e6fa11-cf0f-4f5f-9265-d065f6456a95.png">

**Comment by IslandRhythms** [comment]
My mistake

```javascript
const mongoose = require('mongoose');


const Cat = mongoose.model('Cat', {
    name: String, 
  
    friend: {
      type: mongoose.ObjectId,
      ref: 'Cat'
    }
  });

async function run() {

    await mongoose.connect('mongodb://localhost:27017/cats', {
  useNewUrlParser: true, useUnifiedTopology: true});

  await mongoose.connection.dropDatabase();

  const friend = new Cat({ name: 'Floof' });
    await friend.save();

    const myCat = new Cat({ name: 'Lord Fluffles', friend: friend.id })
    await myCat.save();

    let cats = await Cat.find({}, null, { populate: 'friend' });

    console.log('--populated-friends--', ...cats
  .map(cat => cat.friend)
  .filter(friend => friend));

}

run();

```

**Comment by IslandRhythms** [comment]
@ThoseGrapefruits This issue has been resolved in the 6.3.2 release. Reply if it is not working for you.
